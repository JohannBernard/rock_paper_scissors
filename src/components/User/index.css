.my_coach_user{
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 30vw;
}

.my_coach_user_name {
  line-height: 48px;
  font-size: 48px;
  margin:32px 0 16px;
}

.my_coach_user_score{
  line-height: 48px;
  font-size: 48px;
  font-weight: 600;
  margin: 0;
}

/* Responsive */
@media screen and (max-width: 800px) {
  .my_coach_user_name {
    line-height: 24px;
    font-size: 24px;
  }
  
  .my_coach_user_score{
    line-height: 32px;
    font-size: 32px;
  }
}