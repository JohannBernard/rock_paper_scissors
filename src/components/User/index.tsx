import './index.css'
import IUser from '../../interfaces/IUser'

function User(props: IUser) {
    return (
        <div
            className="my_coach_user"
            data-testid={props.id ? 'testUser-' + props.id : ''}
        >
            <p
                className="my_coach_user_name"
                data-testid={props.id ? 'testUserName-' + props.id : ''}
            >
                {props.name}
            </p>
            <p
                className="my_coach_user_score"
                data-testid={props.id ? 'testUserScore-' + props.id : ''}
            >
                {props.score}
            </p>
        </div>
    )
}

export default User
