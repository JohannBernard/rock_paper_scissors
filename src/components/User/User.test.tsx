import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import User from './index'

afterEach(() => {
    cleanup()
})

test('should render user component', () => {
    render(<User name="Joueur" score={2} id={1} />)
    const itemElement = screen.getByTestId('testUser-1')
    expect(itemElement).toBeInTheDocument()
    const itemIconElement = screen.getByTestId('testUserName-1')
    expect(itemIconElement).toBeInTheDocument()
    expect(itemIconElement).toHaveTextContent('Joueur')
    const itemLabelElement = screen.getByTestId('testUserScore-1')
    expect(itemLabelElement).toHaveTextContent('2')
})

test('matches user snapshot', () => {
    const tree = renderer
        .create(<User name="Joueur" score={2} id={1} />)
        .toJSON()
    expect(tree).toMatchSnapshot()
})
