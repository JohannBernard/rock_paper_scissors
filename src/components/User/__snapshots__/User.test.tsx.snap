// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches user snapshot 1`] = `
<div
  className="my_coach_user"
  data-testid="testUser-1"
>
  <p
    className="my_coach_user_name"
    data-testid="testUserName-1"
  >
    Joueur
  </p>
  <p
    className="my_coach_user_score"
    data-testid="testUserScore-1"
  >
    2
  </p>
</div>
`;
